scripts/src/lib/generator/TemplateGenerator.ts

```diff
@@ -0,0 +1,75 @@
+import JSZip from 'jszip';
+import FileSaver from 'file-saver';
+import { addGradleWrapper } from './GradleWrapper';
+import { addGroovyGradle } from './GradleGroovy';
+import { getKotlinAdapterVersions, getLoaderVersions, getMinecraftYarnVersions } from '../Api';
+import { addModJson } from './ModJson';
+
+export interface Options {
+    minecraftVersion: string,
+    projectName: string,
+    packageName: string,
+    useKotlin: boolean,
+    dataGeneration: boolean,
+    githubActions: boolean,
+    splitSources: boolean,
+}
+
+export interface KotlinOptions {
+	kotlinVersion: string,
+	fabricKotlinAdapterVersion: string
+}
+
+// Computed options are not presented to the user.
+export interface ComputedOptions extends Options {
+	modid: string,
+	loaderVersion: string,
+	yarnVersion: string,
+	kotlin: KotlinOptions | undefined
+}
+
+export interface TemplateOptions {
+	minecraftVersion: string, loaderVersion: string, yarnVersion: string
+}
+
+export async function generateTemplate(options: Options) {
```

Comment:
Massive scope creep, so feel free to disregard:

I think this should be refactored out into it's own package so a cli tool could reuse this. Given the only real dependencies of the generator are `jszip` and `eta` it'd be pretty simple, although this would probably mean introducing an xml parser for node/deno.